src/main/java/com/atlassian/db/replica/spi/ConnectionPoolPerReplicaProvider.java

```diff
@@ -0,0 +1,8 @@
+package com.atlassian.db.replica.spi;
+
+import com.atlassian.db.replica.api.jdbc.JdbcUrl;
+
+@FunctionalInterface
+public interface ConnectionPoolPerReplicaProvider {
```

Comment:
Do we really need to know about the connection pool? We can simplify all of that by:
ReplicaUrl -> Connection

Pools, caches are implementation detail IMO